GDJS/GDJS/Extensions/Builtin/AdvancedExtension.cpp

```diff
@@ -69,6 +69,33 @@ AdvancedExtension::AdvancedExtension() {
                booleanCode + "; }";
       });
 
+  GetAllActions()["SetReturnObject"]
+      .GetCodeExtraInformation()
+      .SetCustomCodeGenerator([](gd::Instruction& instruction,
+                                 gd::EventsCodeGenerator& codeGenerator,
+                                 gd::EventsCodeGenerationContext& context) {
+        gd::String objectsPickingCode;
+
+        for (const auto& objectName : codeGenerator.ExpandObjectsName(
+                 instruction.GetParameter(0).GetPlainString(), context)) {
+          const gd::String& objectNameString = codeGenerator.ConvertToStringExplicit(objectName);
+          const gd::String& objectList = codeGenerator.GetObjectListName(objectName, context);
+          objectsPickingCode += 
+            "{"
+            "const lists = eventsFunctionContext.getObjectsLists("+objectNameString+");"
+            // Clear picked objects list...
+            "for (const list of Object.values(lists.items)) list.length = 0;"
```

Comment:
nit: Do you think we could avoid the Object.values here? It's forcing to allocate then immediately drop an array.